GameLogic/engine/StandardGame.cs

```diff
@@ -6,180 +6,139 @@
-            }
 
-            // Extract only one king cell.
-            var kingCell = kingCells[0];
+            var kingCell = gameState.Board.GetCellsWithPieces(filterByColor: gameState.ActiveColor,
+                                                              filterByPieceType: PieceType.King) // Extract king location.
+                                          .First();
 
-            foreach (var cell in FindAllCellsUnderThreat(gameState))
-            {
-                if (cell.Equals(kingCell)) return true;
-            }
-            return false;
+            return FindAllCellsUnderThreat(filterByColor: gameState.EnemyColor).Any(cell => kingCell == cell);
         }
 
-
-        /// <summary>
-        /// TODO: not finished.
-        /// </summary>
-        public StandardGameState? MakeMove(StandardGameState gameState, Move move)
+        // Applies the move.
+        //
+        // Parameters
+        // ----------
+        // move: The move to apply.
+        // 
+        // Returns
+        // -------
+        // A new instance of StandardGame, if the move is valid. Otherwise, returns null.
+        public StandardGame? MakeMove(Move move)
         {
             var nextGameState = MoveApplier.GetNextGameState(gameState, move);
-            return (IsGameStateValid(nextGameState)) ? nextGameState : null;
+            var nextGame = new StandardGame(nextGameState);
+            return nextGame.IsValid() ? nextGame : null;
         }
 
-        /// <summary>
-        /// TODO: not finished.
-        /// </summary>
-        public bool IsGameStateValid(StandardGameState gameState)
+        // Check if the current game state is valid.
+        //
+        // Returns
+        // -------
+        // true, if the current game state is valid. Otherwise, returns false.
+        public bool IsValid()
         {
             var onlyOneEnemyKing = gameState.Board.GetCellsWithPieces(filterByColor: gameState.EnemyColor,
                                                                       filterByPieceType: PieceType.King).Count == 1;
-            return onlyOneEnemyKing && !IsCheck(gameState);
+
+            var onlyOneKing = gameState.Board.GetCellsWithPieces(filterByColor: gameState.ActiveColor,
+                                                                 filterByPieceType: PieceType.King).Count == 1;
+
+            var enemyKingCell = gameState.Board.GetCellsWithPieces(filterByColor: gameState.EnemyColor,
+                                                                   filterByPieceType: PieceType.King) // Extract king location.
+                                               .First();
+            var checkToEnemyKing = FindAllCellsUnderThreat(filterByColor: gameState.ActiveColor).Any(cell => enemyKingCell == cell);
+
+            // Pawns cannot be located on first and last lines of the height (y-axis).
+            var invalidHeights = new List<int> { 0, 7 };
```

Comment:
I believe it should be renamed smth like `invalidPawnRanks` or `invalidRanksForPawn`

Reply:
Fixed!